Sources/Starknet/Data/MerkleTree.swift

```diff
@@ -0,0 +1,40 @@
+import Foundation
+
+public struct MerkleTree {
+    public let leaves: [Felt]
+    public let hashMethod: HashMethod
```

Comment:
Should we keep `hashMethod` or only use it in initializer and drop afterwards?

The only reason to keep it to keep it seems to me is allowing potential proof-related extensions.